polkadot/node/core/pvf/src/host.rs

```diff
@@ -981,6 +988,52 @@ fn check_landlock(
 	}
 }
 
+/// Check if seccomp is supported and emit a warning if not.
+///
+/// We do this check by spawning a new process and trying to sandbox it. To get as close as possible
+/// to running the check in a worker, we try it... in a worker. The expected return status is 0 on
+/// success and -1 on failure.
+async fn check_seccomp(
+	#[cfg_attr(not(target_os = "linux"), allow(unused_variables))]
+	prepare_worker_program_path: &Path,
+) -> bool {
+	cfg_if::cfg_if! {
+		if #[cfg(target_os = "linux")] {
```

Comment:
we should also only do this for x86.
Maybe we should conditionally compile the entire `seccomp` module to avoid this.

Installing a filter that was compiled for x86 on an arm machine will not fail. It'll install it and kill the process on the first syscall, regardless of what syscall it is or the rest of the filter

Reply:
As we discussed, for this simple filter we could enable it for aarch64 also:

> using the libc syscall constants will transparently switch to the right numbers when building for ARM. and we just need to supply the right arch to seccompiler when building the filter.

But let's leave this gated on `x86_64` until we have the CI job - added a TODO in the code.